### DISCUSSION TITLE
Preventing a JavaScript redirect to a 3rd-party site using intercept

### DISCUSSION THREAD

**Discussion by Undistraction** [cypress-io/cypress#9400] [Questions and Help]
I need to test an auth flow that involves a redirect to a third-party site. I have successfully used the following code to test the behaviour:

```
cy.intercept({ pathname: `/sessions/new` }).as(`loginRedirect`)

cy.visit(`/dashboard/`)

cy.location().then(($location) => {
  cy.wait(`@loginRedirect`).then(($interceptor) => {
    const { query } = urlParse($interceptor.request.url)
    expect(query).to.equal(`?token=true&redirect=${$location.href}`)
  })
})
```

However on our CI we are bumping into  this issue: https://github.com/cypress-io/cypress/issues/4220. Given that I only need to test that the redirect happens, and I don't actually need the page to redirect, I could avoid this issue if there was a way to cancel the redirect behaviour. 

Is there a way to cancel the redirect using `intercept`?

[Update] It seems that even this solution is unreliable, as it is possible for the redirect to happen in between `cy.visit` and `cy.wait`, meaning the test is left waiting for something that already happened.

**Comment by bahmutov** [comment]
See the redirect spec in this recipe https://github.com/cypress-io/cypress-example-recipes/pull/598

(note that this is a PR to fix spying on the redirect call, it should be released in 6.1.0. Stubbing the redirect should work already in 6.0.1)

**Reply by Undistraction** [reply]
@bahmutov Thanks for this. It will be really helpful.
